doc/port_link_training/port-link-training-design.md

```diff
@@ -0,0 +1,336 @@
+	; field                 = value
+	...
+	link_training           = STRING             ; link training configuration
+
+Valid value of the new fields are described below:
+
+- link_training:  
+String value, the administratively specified port link training configuration.
+When "link_training" is not specified, the port link training should be disabled  
+
+ | Mode |     Description                                                 |
+ |:----:|:---------------------------------------------------------------:|
+ | auto | Enable link-training if applicable to the transceiver attached  |
+ | on   | Enable link-training regardless of the transceiver type         |
+ | off  | Disable link-training                                           |
+
+For deployment considerations, users are encouraged to use only **auto** and **off** for the **link_training**, while the **on** mode is for test purpose, it should only be activated when the transceiver is not correctly identified by the pmon#xcvrd, and forcing link-training enabled is intended.
+
+#### Application DB Enhancements
+
+The following fields will be introduced into **PORT_TABLE** table:
+
+	; Defines information for port configuration
+	key                     = PORT_TABLE:port_name ; configuration of the port
+	; field                 = value
+	...
+	link_training           = STRING               ; operational link training config
+	link_training_status    = STRING               ; operational link training status
+	link_training_failure   = STRING               ; operational link training failure status
+	link_training_rxstatus  = STRING               ; operational link training rx status
+	xcvr_capabilities       = STRING               ; transceiver capabilities
+
+- link_training:  
+String value, the admin port link training config.
+- link_training_status:  
+String value, the operational port link training status. "on" indicates port link training is enabled, otherwise disabled.
+- link_training_failure:  
+String value, the operational port link training failure status. "none" indicates no error is detected, otherwise any of the link training failure status defined in SAI. For example: "none", "lock", "snr" and "timeout".
+- link_training_rxstatus:  
+String value, the operational port link training rx status. any of the link training rx status defined in SAI. For example: "trained", "not-trained".
+- xcvr_capabilities:
+String value, the transceiver capabilities provided by pmon#xcvrd, it's a set of capabilities separated by comma, where "LT" stands for "link training". See detail description in section [PMON xcvrd Considerations](#pmon-xcvrd-considerations)
+
+To minimize system overhead, instead of periodically fetching the operational link training status, the corresponding fields in the APPL_DB will only be updated in the following events
+
+- Per-Port link status changes  
+In this case, only the **link_training_status** will be updated.
```

Comment:
How does it get link training status? By which SAI attribute?

Reply:
The operation state could be fetch by the following SAI attributes
- SAI_PORT_ATTR_LINK_TRAINING_FAILURE_STATUS
- SAI_PORT_ATTR_LINK_TRAINING_RX_STATUS
And 'link_training_status' will be updated to display only one of the best appropriate description for the current state.
e.g. 'not_trained' will only be presented when none of failure is reported by SAI_PORT_ATTR_LINK_TRAINING_FAILURE_STATUS, and the LT hardware engine failed to get a fine-tuned pre-emphasis

Reply:
The HLD will later be updated to have this information, it should be done later today